SCClassLibrary/Common/Files/SoundFile.sc

```diff
@@ -70,6 +71,36 @@ SoundFile {
 		^res
 	}
 
+	*write {|pathName, array, headerFormat, sampleRate, sampleFormat|
+        var file = SoundFile.new;
+
+        if(array.isKindOf(SequenceableCollection).not,
+            {Error("Argument 'array' must be an instance of SequenceableCollection (e.g., Array). Received" + array.class.asString + "with value" + array.asString).throw});
+
+        if(array.shape.size > 2,
+            {Error(
+                "Cannot write an array with nested channels."
+                + "The array should be formatted as: "
+                + "[  [channel 1 samples...], [channel 2 samples...], ...]"
+            ).throw}
+        );
+
+        headerFormat !? file.headerFormat_(_);
+        sampleFormat !? file.sampleFormat_(_);
+        sampleRate !? file.sampleRate_(_);
+
+        file.numChannels_(if(array.shape.size == 1, 1, {array.shape[0]}));
+
+        if(file.openWrite(pathName).not,
+            {Error("Could not open file at path:" + pathName).throw});
+
+        // interlace the samples and write
+        file.writeData(FloatArray.with(*array.lace(array.shape.reduce('*')).asFloat));
```

Comment:
I'm not sure if this conversion with `lace` will turn out to be a curse later on. At least it is very implicit.

Reply:
After a quick benchmark, flop.flat is about 50x slower.
```supercollider
d = [];

[10, 100, 1000, 10000, 100000].do{|n|
	a = {{0.0.rrand(1.0)} ! n} ! 2;
	10.do{
		var dur_lace = {
			a.lace(a.shape.reduce('*'));
			//FloatArray.with(*a)
		}.bench();
		
		var dur_flop = {
			a.flop.flat;
			//a.as(FloatArray)
		}.bench();
		
		d = d.add(dur_flop / dur_lace ); 
	}
};

d.mean
```

Reply:
Ah good to know! I had thought that flat is a primitive like flop. The advantage of `flop` is that you are guaranteed to have the right shape on the outermost level.

Also I've added a comment, because I mistook the documentation. 

> Also, this doesn't check that the shape of the array is regular, meaning all nested arrays are the same size. Do you think it is worth adding a check for that?

If the result would be a corrupted sound file, yes.

Reply:
By using lace the values will wrap around to the beginning, meaning the sound file is always valid and the shape of the array is guaranteed, but the user probably didn't mean for this to happen. Should this throw an error?

Here would be the check, unless there is some form of 'isRegularShape' method on array that I haven't heard of....
```supercollider
if(array.rank == 2) {
	var first_size = array[0].size();
	array.do{|channel, index|
		if (channel.size() != first_size) {
			Error("All channels should have the same size, first channels's size was" + first_size + "but channel number" + index + "had a size of" + channel.size()).throw
		}
	}
}
```

Reply:
I've put the lace back at the end as well because mutating it changed the number of channels making the rest of the logic stateful and confusing.

Reply:
Is this okay as it is now?